stock_restrict_lot/models/stock_move.py

```diff
@@ -88,3 +89,56 @@ def _split(self, qty, restrict_partner_id=False):
         if vals_list and self.restrict_lot_id:
             vals_list[0]["restrict_lot_id"] = self.restrict_lot_id.id
         return vals_list
+
+    def get_all_dest_moves(self):
+        res = self.move_dest_ids
+        moves_to_search = self.move_dest_ids
+        while moves_to_search:
+            new_dest_ids = moves_to_search.move_dest_ids
+            moves_to_search = new_dest_ids - res
+            res |= new_dest_ids
+        return res
+
+    def get_all_orig_moves(self):
+        res = self.move_orig_ids
+        moves_to_search = self.move_orig_ids
+        while moves_to_search:
+            new_orig_ids = moves_to_search.move_orig_ids
+            moves_to_search = new_orig_ids - res
+            res |= new_orig_ids
+        return res
+
+    def write(self, vals):
+        if "restrict_lot_id" not in vals:
+            return super().write(vals)
+        else:
+            restrict_lot_id = vals.pop("restrict_lot_id")
+            restrict_lot = self.env["stock.lot"].browse(restrict_lot_id)
+            chained_moves = self | self.get_all_dest_moves() | self.get_all_orig_moves()
+            if any(
+                [
+                    sm.state == "done" and sm.lot_ids != restrict_lot
+                    for sm in chained_moves
+                ]
+            ):
+                raise ValidationError(
+                    _(
+                        "You can't modify the Lot/Serial number "
+                        "because at least one move in the chain has "
+                        "already been done with another Lot/Serial number."
+                    )
+                )
+            super(StockMove, chained_moves).write({"restrict_lot_id": restrict_lot_id})
+        return super().write(vals)
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
What is the case where the lot is not propagated during creation ?
Is it a push rule case or something ? 
Wouldn't it be better to manage it there specifically instead on this create override ?

Reply:
You're right, this is happening with push rules.

Do you mean overriding the _run_push method for "stock.rule" ? I can probably do that...

Reply:
It seems better to me to override for instance `_push_prepare_move_copy_values` indeed, rather than overriding the create method, if it only fix the push rule case.

Reply:
Good idea, it's done.